luigi/scheduler.py

```diff
@@ -862,9 +862,8 @@ def _upstream_status(self, task_id, upstream_status_table):
                     elif upstream_status_table[dep_id] == '' and dep.deps:
                         # This is the postorder update step when we set the
                         # status based on the previously calculated child elements
-                        status = max((upstream_status_table.get(a_task_id, '')
-                                      for a_task_id in dep.deps),
-                                     key=UPSTREAM_SEVERITY_KEY)
+                        upstream_severities = list(upstream_status_table.get(a_task_id) for a_task_id in dep.deps if a_task_id in upstream_status_table) or ['']
+                        status = min(upstream_severities, key=UPSTREAM_SEVERITY_KEY)
```

Comment:
Here is the case;

When any `SUCCESS` task exists among upstream tasks, `upstream_status_table.get(a_task_id, '')` results `''` empty string. Because there is no record in `upstream_status_table` for `SUCCESS` tasks. This will cause the wrapper is as PENDING all the time even there is no PENDING task among its upstream tasks. This will also let worker runs forever. 

Here are some SS showing two case;

### Case which is just swtiching max to min;
![screen shot 2016-07-28 at 11 11 53](https://cloud.githubusercontent.com/assets/1279644/17205761/01b303ba-54b5-11e6-99f6-c1f261ca7a67.png)

As you can se the wrapper task which is `TaskLevelRetryPolicy` runs all the time, even its dependencies `SUCCESS` and `DISABLED`. This is happenning because there is at least one `SUCCESS` task.

### The case what is in my commit;
![screen shot 2016-07-28 at 11 14 51](https://cloud.githubusercontent.com/assets/1279644/17205813/5356489e-54b5-11e6-9ee7-90f4d678ade8.png)

In this case, the empty string upstream severity is ignored. First, it tries to find min among elements not includes empty string. Even there is SUCCESS which causes empty string return,
elmination will be done among severities. You can see it on the SS.